src/main/webapp/app/course/dashboards/assessment-dashboard/assessment-dashboard.component.ts

```diff
@@ -219,10 +219,11 @@ export class AssessmentDashboardComponent implements OnInit {
                 exercises,
                 (exercise) =>
                     // finds if all assessments for all correctionRounds are finished
-                    exercise.numberOfAssessmentsOfCorrectionRounds
+                    ((exercise.numberOfAssessmentsOfCorrectionRounds
                         ?.map((round) => round.inTime === exercise.numberOfSubmissions?.inTime)
                         .reduce((acc, current) => acc && current) &&
-                    exercise.totalNumberOfAssessments?.inTime === exercise.numberOfSubmissions?.inTime &&
+                        exercise.totalNumberOfAssessments?.inTime === exercise.numberOfSubmissions?.inTime) ||
+                        exercise.allowComplaintsForAutomaticAssessments) &&
```

Comment:
This is really hard to read, maybe you could extract some parts into extra variables or find another solution?